monai/data/utils.py

```diff
@@ -346,9 +347,21 @@ def list_data_collate(batch: Sequence):
             ret = {}
             for k in elem:
                 key = k
-                ret[key] = default_collate([d[key] for d in data])
+                data_for_batch = [d[key] for d in data]
+                ret[key] = default_collate(data_for_batch)
+                if isinstance(ret[key], MetaTensor) and all(isinstance(d, MetaTensor) for d in data_for_batch):
```

Comment:
May I know why need to collate `affine`, `meta`, `transforms` explicitly? We don't have such logic for `image_meta_dict` now.

Thanks.